src/Level.cpp

```diff
@@ -202,6 +202,33 @@ void Level::display(
 /**
  *
  */
+void Level::displayWithTransform(
+    const std::shared_ptr<utils::Context>& context,
+    const unsigned short& floor
+) const
+{
+    /* calculate the index of the first cell of the given floor */
+    const unsigned short firstCellIndex = floor * 256;
+
+    /* calculate the index of the last cell of the given floor */
+    const unsigned short lastCellIndex = floor * 256 + 256;
+
+    /* only display the cells of the given floor; all the other cells of the
+       level are ignored */
+    for(
+        unsigned short index = firstCellIndex;
+        index < lastCellIndex;
+        index++
+    )
+    {
+        /* get an unique pointer, get the cell object from this pointer */
+        (*cells[index]).displayWithTransform(context, commonTransform);
+    }
+}
```

Comment:
the common transform should be a nullable pointer, the transform is created by the animation and used here only if not null